### ISSUE TITLE
the timeout option on the goTo command can't be disabled

### ISSUE THREAD

**Issue by armanm** [puppeteer/puppeteer#965]
**What steps will reproduce the problem?**

I'm on puppeteer 0.10.0, OSX trying to take some screenshots. The `goto` command is supposed to [allow disabling `timeout`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagegotourl-options) by giving it a value `0` but that doesn't seem to work:

**What is the expected result?**

I expect it to wait indefinitely for navigation to complete.

**What happens instead?**

```
(node:57158) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Navigation Timeout Exceeded: 0ms exceeded
(node:57158) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: Navigation Timeout Exceeded: 0ms exceeded
Error: Navigation Timeout Exceeded: 0ms exceeded
    at NavigatorWatcher.<anonymous> (/Users/arman/src/sites/node_modules/puppeteer/node6/NavigatorWatcher.js:100:13)
    at next (native)
    at step (/Users/arman/src/sites/node_modules/puppeteer/node6/NavigatorWatcher.js:44:24)
    at Promise.resolve.then.value (/Users/arman/src/sites/node_modules/puppeteer/node6/NavigatorWatcher.js:55:15)
```

**Comment by vsemozhetbyt** [comment]
https://github.com/GoogleChrome/puppeteer/pull/887 merged only recently, so v0.10.0 does not support this yet: https://github.com/GoogleChrome/puppeteer/blob/v0.10.0/docs/api.md#pagegotourl-options

Try to update to the v0.12.0-alpha by `npm install https://github.com/GoogleChrome/puppeteer/`

**Comment by Xirax** [comment]
Seems not to work form me on 1.19.0 puppeteer :cactus:
